nodeadm/internal/configprovider/userdata.go

```diff
@@ -17,22 +18,41 @@ import (
 
 const (
 	contentTypeHeader          = "Content-Type"
+	contentEncodingHeader      = "Content-Encoding"
 	mimeBoundaryParam          = "boundary"
 	multipartContentTypePrefix = "multipart/"
 	nodeConfigMediaType        = "application/" + api.GroupName
 )
 
-type userDataConfigProvider struct{}
+type userDataProvider interface {
+	GetUserData() ([]byte, error)
+}
+
+type imdsUserDataProvider struct{}
+
+func (p *imdsUserDataProvider) GetUserData() ([]byte, error) {
+	return imds.GetUserData()
+}
+
+type userDataConfigProvider struct {
+	userDataProvider userDataProvider
+}
```

Comment:
Why do you need this wrapper?

Reply:
This is for testing, it allows us to mock out the IMDS calls